packages/web/src/pages/VaultEditor/vaultEditorService.ts

```diff
@@ -140,33 +145,62 @@ export async function sendEditionToGovApproval(editSessionId: string): Promise<I
  * @param vaultAddress - The vault address
  * @param chainId - The chain id of the vault
  */
-export async function getVaultInformation(vaultAddress: string, chainId: number): Promise<IVaultStatusData> {
+export async function getVaultInformation(vault: IVault | undefined): Promise<IVaultStatusData | undefined> {
+  if (!vault) return undefined;
+  if (vault.version === "v1") return undefined;
+  if (vault.version === "v3" && !vault?.claimsManager) return undefined;
+
   const vaultContractInfo = {
-    address: vaultAddress as `0x${string}`,
-    abi: HATSVaultV2_abi,
-    chainId,
+    address: vault.id as `0x${string}`,
+    abi: vault.version === "v2" ? HATSVaultV2_abi : HATSVaultV3_abi,
```

Comment:
better to be explicit here 
```suggestion
    if (vault.version === "v2") {
  abi = HATSVaultV2_abi;
} else if (vault.version === "v3") {
  abi = HATSVaultV3_abi;
} else {
  // Handle cases where vault.version is neither "v2" nor "v3"
  console.error("Unsupported vault version:", vault.version);
  // You might throw an error or handle it differently based on your needs
}
```